lib/_tls_wrap.js

```diff
@@ -869,9 +869,25 @@ Server.prototype.setOptions = function(options) {
   }
 
   if (options.pfx) this.pfx = options.pfx;
-  if (options.key) this.key = options.key;
+  var defaultCiphers = options.ciphers === tls.DEFAULT_CIPHERS;
+  if (!options.key) {
+    if ((options.ciphers === undefined || defaultCiphers) && !options.pfx) {
+      throw new Error('key is a required parameter for Server.createServer');
+    }
+  } else {
+    this.key = options.key;
+  }
+
   if (options.passphrase) this.passphrase = options.passphrase;
-  if (options.cert) this.cert = options.cert;
+
+  if (!options.cert) {
+    if ((options.ciphers === undefined || defaultCiphers) && !options.pfx) {
+      throw new Error('cert is a required parameter for Server.createServer');
+    }
+  } else {
+    this.cert = options.cert;
+  }
+
   if (options.ca) this.ca = options.ca;
```

Comment:
@cbshakumar you mean if cert authority is provided then skip this whole cert/key error handling right?